### PR TITLE
templates/components: Add data-control-name attribute on row checkbox

### PR DESCRIPTION
**Pull request by jlevesy**
### What does this PR do?

This PR adds a `data-control-name=` attributes where needed.

Related to : https://github.com/upfluence/backlog/issues/657

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

**Review by phndiaye — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "ember-sortable": "^1.12.9",
     "ember-truth-helpers": "^2.1.0",
     "tether": "^1.4.7",
-    "upfluence-oss-components": "^2.x"
+    "upfluence-oss-components": "^2.0.31"
```
Comment (JulienVannier66) [inline package.json:31]: Stay on "^2.x", no?
You can do that with this command `yarn upgrade upfluence-oss-components@^2.x` and it gets last version on master automatically

Comment (jlevesy) [reply package.json]: I'm dropping that change then.
`yarn.lock` is in the gitignore for some reason, which is weird AF.

File: app/templates/components/hyper-table.hbs [hunk 2]
```diff
@@ -7,9 +7,13 @@
...
-      {{input
-        type="text" placeholder="Search..." value=_searchQuery
-        class="form-control upf-input upf-input--small"}}
+      <Input
+        type="text"
+        placeholder="Search..."
+        @value={{_searchQuery}}
+        data-control-name="table_search_input"
+        class="form-control upf-input upf-input--small"
+        />
```
Comment (phndiaye) [inline app/templates/components/hyper-table.hbs:16]: can you add a `@` before `type` and `placeholder` too ?

File: app/templates/components/hyper-table.hbs [hunk 3]
```diff
@@ -61,9 +73,10 @@
 
       <div class="available-fields-wrapper__fields">
         <div class="search">
-          {{input
-            type="text" value=_availableFieldsKeyword
-            class="upf-input upf-input--small" placeholder="Search Fields"}}
+          <Input
+            type="text" @value={{_availableFieldsKeyword}}
```
Comment (phndiaye) [inline app/templates/components/hyper-table.hbs:77]: same as above: `@type="text"`

File: app/templates/components/hyper-table.hbs [hunk 4]
```diff
@@ -52,7 +63,8 @@
           <div><i class="fa fa-caret-right"></i></div>
         </div>
         {{#each _fieldCategories as |category|}}
-          <div class="field-category {{if (eq _activeFieldCategory category.key) 'field-category--active'}}" {{action "setFieldCategory" category.key}}>
+          <div class="field-category {{if (eq _activeFieldCategory category.key) 'field-category--active'}}"
+               data-control-name="field_category_toggle_{{ category.key }}" {{action "setFieldCategory" category.key}}>
```
Comment (phndiaye) [inline app/templates/components/hyper-table.hbs:67]: can you use `concat` here too ?

File: app/templates/components/hyper-table.hbs [hunk 5]
```diff
@@ -122,7 +136,7 @@
           {{#if manager.options.features.selection}}
             <div class="hypertable__column hypertable__column--selection {{concat "hypertable__column--size-" column.size}}">
               <header>
-                {{upf-checkbox value=_allRowsSelected size="sm"}}
+                {{upf-checkbox po value=_allRowsSelected size="sm" data-control-name="row_select_all_toggle_checkbox"}}
```
Comment (phndiaye) [inline app/templates/components/hyper-table.hbs:139]: upf-checkbox po (??)

File: app/templates/components/hyper-table.hbs (hunk 4 above)
Comment (jlevesy) [reply app/templates/components/hyper-table.hbs:67]: I'm not in a component evaluation, how can I ?

File: package.json (hunk 1 above)
Comment (JulienVannier66) [reply package.json]: My bad you right, for this project we don't push `yarn.lock`

File: app/templates/components/hyper-table.hbs (hunk 4 above)
Comment (JulienVannier66) [reply app/templates/components/hyper-table.hbs:67]: Somethink like that : `name={{concat "field_category_toggle_" category.key)`

Comment (jlevesy) [reply app/templates/components/hyper-table.hbs:67]: :exploding_head:

**Review by JulienVannier66 — APPROVED** [review]
LGTM

**Review by phndiaye — APPROVED** [review]

**Pull request merged**